src/prefect/client/schemas/filters.py

```diff
@@ -538,7 +538,9 @@ class DeploymentFilter(PrefectBaseModel, OperatorMixin):
         default=None, description="Filter criteria for `Deployment.work_queue_name`"
     )
     concurrency_limit: Optional[DeploymentFilterConcurrencyLimit] = Field(
-        default=None, description="Filter criteria for `Deployment.concurrency_limit`"
+        default=None,
+        description="DEPRECATED: Prefer `Deployment.concurrency_limit_id` over `Deployment.concurrency_limit`. If provided, will be ignored for backwards-compatibility.",
```

Comment:
Yeah I removed this at first in this PR and then decided to go this safer route.

I could use some guidance on a timeline tho. Partially thought out plan to do some monitoring in cloud for a little to ensure nobody is using and then remove it. It seems very unlikely anybody is using it so there's a decision to be made on removing.

Reply:
The default deprecation period is six months, but I think we could shorten it in this case since it would affect so few versions. Maybe a 2-3 month deprecation period?

Reply:
Sure! added a "Will be removed after January 2025" here and in the server schema.